yarn-project/ethereum/src/contracts/empire_base.ts

```diff
@@ -64,16 +68,6 @@ export async function signVoteWithSig(
     nonce,
   };
 
-  if (!walletClient.account) {
-    throw new Error('Wallet client must be connected to an account');
-  }
-
-  const signatureHex = await walletClient.signTypedData({
-    account: walletClient.account,
-    domain,
-    types,
-    primaryType: 'Vote',
-    message,
-  });
-  return Signature.fromString(signatureHex);
+  const msg = hashTypedData({ domain, types, primaryType: 'Vote', message });
+  return Signature.fromString(await signer(msg));
```

Comment:
I think the new signing is equivalent. The new flow uses the key store that exists in the validator-client to sign the hash of the structured data

Reply:
Yep, seems like it is: https://github.com/wevm/viem/blob/ba8319f71503af8033fd3c77cfb64c7eb235c6a9/src/accounts/utils/signTypedData.ts#L34-L47